quipucords/api/scanjob/model.py

```diff
@@ -274,3 +287,40 @@ def validate_status_change(self, target_status, valid_current_status):
                          target_status, self.status)
             return True
         return False
+
+    def create_product_status_dict(self):
+        """Construct a dictionary based on the disabled products.
+
+        :returns: a dictionary representing the updated collection
+        status of the optional products
+        """
+        # Grab the optional products status dict and create
+        # a default dict (all products default to True)
+        product_status = self.get_optional_products_args()
+        product_default = {self.OPT_JBOSS_EAP: True,
+                           self.OPT_JBOSS_FUSE: True,
+                           self.OPT_JBOSS_BRMS: True}
+
+        # If specified, turn off fact collection for fuse
+        if not product_status[self.ARG_JBOSS_FUSE]:
+            product_default[self.OPT_JBOSS_FUSE] = False
+        # If specified, turn off fact collection for brms
+        if not product_status[self.ARG_JBOSS_BRMS]:
+            product_default[self.OPT_JBOSS_BRMS] = False
+        # If specified and both brms & fuse are false
+        # turn off fact collection for eap
+        if not product_status[self.ARG_JBOSS_EAP] and \
+                (not product_default[self.OPT_JBOSS_FUSE]) and \
+                (not product_default[self.OPT_JBOSS_BRMS]):
+            product_default[self.OPT_JBOSS_EAP] = False
+
+        return product_default
+
+    def get_optional_products_args(self):
+        """Access disabled_optional_products as a dict instead of a string.
+
+        :returns: python dict containing the status of optional products
+        """
```

Comment:
With my suggestion of just storing null you could just add a step here that checks for `None` and returns and empty dictionary so your `create_product_status_dict()` method doesn't need to change.